hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java

```diff
@@ -736,13 +764,14 @@ public SqmCorrelatedRoot<?> visitCorrelatedRoot(SqmCorrelatedRoot<?> correlatedR
 
 		@Override
 		public SqmBagJoin<?, ?> visitBagJoin(SqmBagJoin<?, ?> join) {
+			final SqmFrom lhsCopy = findSqmFromCopy( join.getLhs() );
```

Comment:
Maybe I'm not seeing it, but why did you move the `findSqmFromCopy` call up here?

Reply:
Because we could copy a join with a lhs that was still not copied (e.g. another join), thus triggering the copy of that as well. That would in turn trigger a new copy of the original join (since that was still being constructed and not yet registered) thus duplicating the object and triggering an "Alias [x] used for multiple from-clause elements" error.

Putting `findSqmFromCopy` of the lhs before checking the `sqmFromCopyMap` allows us to always only copy the join once and reuse the instance correctly.

Reply:
Ok, I guess the name `findSqmFromCopy` confused me 😅 
Would you mind trying to use the following implementation for `split` instead?

```
for ( EntityDomainType<?> mappedDescriptor : unmappedPolymorphicDescriptor.getImplementors() ) {
	i++;
	final SqmCopyContext copyContext = SqmCopyContext.simpleContext();
	copyContext.registerCopy( 
		unmappedPolymorphicFromElement,
		new SqmRoot<>(
					mappedDescriptor,
					unmappedPolymorphicFromElement.getExplicitAlias(),
					unmappedPolymorphicFromElement.isAllowJoins(),
					unmappedPolymorphicFromElement.nodeBuilder()
		)
 	);
	expanded[i] = statement.copy( copyContext );
}
```

I think this will allow us to get rid of the entire `UnmappedPolymorphismReplacer`.

Reply:
I've updated the PR to use the existing SQM `copy` logic as suggested, avoiding the copy of parameters.

I've also removed the need for an `instanceof SqmPath` check in the copy context by always copying SqmFrom's `lhs` before checking if the current object was already copied. Let me know what you think.